cmd/update/volume_move_replica.go

```diff
@@ -0,0 +1,205 @@
+package update
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"code.storageos.net/storageos/c2-cli/apiclient"
+	"code.storageos.net/storageos/c2-cli/cmd/argwrappers"
+	"code.storageos.net/storageos/c2-cli/cmd/clierr"
+	"code.storageos.net/storageos/c2-cli/cmd/flagutil"
+	"code.storageos.net/storageos/c2-cli/cmd/interfaces"
+	"code.storageos.net/storageos/c2-cli/cmd/runwrappers"
+	"code.storageos.net/storageos/c2-cli/model"
+	"code.storageos.net/storageos/c2-cli/pkg/health"
+	"code.storageos.net/storageos/c2-cli/pkg/id"
+	"code.storageos.net/storageos/c2-cli/pkg/labels"
+	"code.storageos.net/storageos/c2-cli/pkg/version"
+)
+
+type volumeMoveReplicaCommand struct {
+	config  interfaces.ConfigProvider
+	client  interfaces.Client
+	display interfaces.Displayer
+
+	namespace string
+	volumeID  string
+	labels    labels.Set
+
+	// cmd values
+	fromNode string
+	toNode   string
+
+	// useCAS determines whether the command makes the request constrained by
+	// the provided casVersion.
+	useCAS     func() bool
+	casVersion string
+	useAsync   bool
+
+	writer io.Writer
+}
+
+func (c *volumeMoveReplicaCommand) runWithCtx(ctx context.Context, cmd *cobra.Command, args []string) error {
+
+	useIDs, err := c.config.UseIDs()
+	if err != nil {
+		return err
+	}
+
+	// fetch ns
+
+	var nsID id.Namespace
+
+	if useIDs {
+		nsID = id.Namespace(c.namespace)
+	} else {
+		ns, err := c.client.GetNamespaceByName(ctx, c.namespace)
+		if err != nil {
+			return err
+		}
+		nsID = ns.ID
+	}
+
+	// fetch volume
+
+	vol, err := c.getVolume(ctx, nsID)
+	if err != nil {
+		return err
+	}
+
+	// check the volume allows for the move to take place
+	if len(vol.Replicas) == 0 {
+		return errors.New("volume has no replicas")
+	}
```

Comment:
We could probably add some more sanity checks, in a similar command I had the following checks:
```
	if vol.Master.Health != health.MasterOnline {
		return fmt.Errorf("cannot move volume - master unhealthy: %v", vol.Master.Health.String())
	}
	for _, replica := range vol.Replicas {
		if replica.Health != health.ReplicaReady || replica.Promotable == false {
			return fmt.Errorf("cannot move volume - replica unhealthy, id: %v health: %v, promotable: %v", replica.ID, replica.Health, replica.Promotable)
		}
	}
```